taurus/entity/bounds/real_bounds.py

```diff
@@ -93,19 +93,11 @@ def _convert_bounds(self, target_units):
             A tuple of the (lower_bound, upper_bound) in the target units.
 
         """
-        # if neither have units, then just return the bounds
-        if not self.default_units and not target_units:
-            return self.lower_bound, self.upper_bound
-        # if either have units but both don't, then we can't return anything
-        elif not self.default_units or not target_units:
```

Comment:
RealBounds are required to have units now, so there is no need to have these checks.